src/Essentials/src/Platform/WindowStateManager.ios.cs

```diff
@@ -84,23 +81,43 @@ class WindowStateManagerImplementation : IWindowStateManager
 
 		public UIWindow? GetCurrentUIWindow()
 		{
-			// This call site is reachable on: 'iOS' 10.0 and later.0 'UIApplication.KeyWindow.get' is unsupported on: 'ios' 13.0 and later.
-			var window = UIApplication.SharedApplication.KeyWindow;
+			var window = GetKeyWindow();
 
 			if (window != null && window.WindowLevel == UIWindowLevel.Normal)
 				return window;
 
 			if (window == null)
 			{
-				// This call site is reachable on: 'iOS' 10.0 and later. 'UIApplication.Windows.get' is unsupported on: 'ios' 15.0 and later.
-				window = UIApplication.SharedApplication
-					.Windows
+				window = GetWindows()?
 					.OrderByDescending(w => w.WindowLevel)
 					.FirstOrDefault(w => w.RootViewController != null && w.WindowLevel == UIWindowLevel.Normal);
 			}
-#pragma warning restore CA1416
 
 			return window;
 		}
+
+		UIWindow? GetKeyWindow()
+		{
+			if(OperatingSystem.IsIOSVersionAtLeast(13))
```

Comment:
Why is this 13 and the other 15? Seems to be the same code?

Reply:
Great question.  According to the comments in the original file, they following different deprecation lifecycles.  This also backs Apple's documentation.

`UIApplication.SharedApplication.Windows` shows iOS 15
https://github.com/dotnet/maui/blob/81f773f1e37621ce5cee0129595ad8e582fd7f4f/src/Essentials/src/Platform/WindowStateManager.ios.cs#L95-L97
https://developer.apple.com/documentation/uikit/uiapplication/1623104-windows

`UIApplication.SharedApplication.KeyWindow` shows iOS 13
https://github.com/dotnet/maui/blob/81f773f1e37621ce5cee0129595ad8e582fd7f4f/src/Essentials/src/Platform/WindowStateManager.ios.cs#L87-L88
https://developer.apple.com/documentation/uikit/uiapplication/1622924-keywindow

Reply:
ok, if this is expected, then all good.